include/measurement_kit/nettest.hpp

```diff
@@ -0,0 +1,1922 @@
+
+    /// Starts a "TcpConnect" nettest.
+    /// \param settings contains the settings configuring this nettest.
+    /// \param router is the router used to route nettest events. If the
+    /// router is `nullptr`, we'll use an internal dummy router.
+    /// \return true on success, false on failure. There are two reasons why
+    /// this method could fail. Either the provided settings are not JSON
+    /// serializable because there was a string that was not valid UTF-8, or
+    /// you already used this class to start a nettest. In both cases, log
+    /// messages will be routed, so you will know what happened.
+    bool start_tcp_connect(settings::TcpConnectSettings settings, routers::Router *router) noexcept;
+
+    /// Starts a "Telegram" nettest.
+    /// \param settings contains the settings configuring this nettest.
+    /// \param router is the router used to route nettest events. If the
+    /// router is `nullptr`, we'll use an internal dummy router.
+    /// \return true on success, false on failure. There are two reasons why
+    /// this method could fail. Either the provided settings are not JSON
+    /// serializable because there was a string that was not valid UTF-8, or
+    /// you already used this class to start a nettest. In both cases, log
+    /// messages will be routed, so you will know what happened.
+    bool start_telegram(settings::TelegramSettings settings, routers::Router *router) noexcept;
+
+    /// Starts a "WebConnectivity" nettest.
+    /// \param settings contains the settings configuring this nettest.
+    /// \param router is the router used to route nettest events. If the
+    /// router is `nullptr`, we'll use an internal dummy router.
+    /// \return true on success, false on failure. There are two reasons why
+    /// this method could fail. Either the provided settings are not JSON
+    /// serializable because there was a string that was not valid UTF-8, or
+    /// you already used this class to start a nettest. In both cases, log
+    /// messages will be routed, so you will know what happened.
+    bool start_web_connectivity(settings::WebConnectivitySettings settings, routers::Router *router) noexcept;
+
+    /// Starts a "Whatsapp" nettest.
+    /// \param settings contains the settings configuring this nettest.
+    /// \param router is the router used to route nettest events. If the
+    /// router is `nullptr`, we'll use an internal dummy router.
+    /// \return true on success, false on failure. There are two reasons why
+    /// this method could fail. Either the provided settings are not JSON
+    /// serializable because there was a string that was not valid UTF-8, or
+    /// you already used this class to start a nettest. In both cases, log
+    /// messages will be routed, so you will know what happened.
+    bool start_whatsapp(settings::WhatsappSettings settings, routers::Router *router) noexcept;
+
+    // Event processing methods
+    // ````````````````````````
```

Comment:
This mimics very closely the FFI API. I figured that deviating for that would have added more complexity.